requests/utils.py

```diff
@@ -825,3 +825,16 @@ def rewind_body(prepared_request):
                                         "body for redirect.")
     else:
         raise UnrewindableBodyError("Unable to rewind request body for redirect.")
+
+def unicode_is_ascii(u_string):
+    """Determine if unicode string only contains ASCII characters.
+
+    :param str u_string: unicode string to check. Must be unicode
+        and not Python 2 `str`.
+    :rtype: bool
+    """
```

Comment:
Ok, let's move this to `_internal_utils`. Also, I apologise about the confusion of `unicode`: let's import `str` from `compat.py` and use it here for an assertion.